core/templates/dev/head/library/Library.js

```diff
@@ -65,9 +65,11 @@ oppia.controller('Library', [
       // Transforms the group names into translation ids
       $scope.leftmostCardIndices = [];
       for (i = 0; i < $scope.libraryGroups.length; i++) {
-        $scope.leftmostCardIndices.push(0);
-        $scope.libraryGroups[i].translationId = i18nIdService.getLibraryId(
-          'groups', $scope.libraryGroups[i].header);
+        if ($scope.libraryGroups[i].show_view_all_button) {
+          $scope.leftmostCardIndices.push(0);
+          $scope.libraryGroups[i].translationId = i18nIdService.getLibraryId(
```

Comment:
If we are using header_i18n_id throughout (see comment above) then we don't need to use i18nIdService?

Also, maybe we should call this $scope.libraryGroups[i].headerI18nId instead of translationId, for consistency.